googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -1363,19 +1367,27 @@ constexpr bool InstantiateTypedTestCase_P_IsDeprecated() { return true; }
     GTEST_CONCAT_TOKEN_(gtest_label_testthrow_, __LINE__): \
       fail(gtest_msg.value)
 
+// We need to declare the ConstCharPtr object in the scope of the if. It
+// can't go into the body of the if because then we cannot access it in the
+// else. It also can't go before the if block because than we have to make up
+// some very unique name (line number is not enough).
 #define GTEST_TEST_NO_THROW_(statement, fail) \
   GTEST_AMBIGUOUS_ELSE_BLOCKER_ \
-  if (::testing::internal::AlwaysTrue()) { \
+  if (::testing::internal::ConstCharPtr gtest_msg = ".") { \
     try { \
       GTEST_SUPPRESS_UNREACHABLE_CODE_WARNING_BELOW_(statement); \
     } \
+    catch (const std::exception& e) { \
+      gtest_msg = std::string(": ") + e.what(); \
```

Comment:
no woulder CI is failing:
`gtest_msg` holds a pointer to a temporary object that is destroyed immediately after this statement - 
hello usage of dangling pointer :)

Original implementation with `AdditionalMessage` didn't have this issue, since it owned a copy of the message, which is fine.

My opinion - we need `AdditionalMessage` class back, maybe named more consistently with `ConstCharPtr`.
And `ConstCharPtr& ConstCharPtr::operator=(const std::string& message)` shouldn't be allowed, by definition of `class ConstCharPtr`.